src/platform/plugins/shared/embeddable/public/react_embeddable_system/react_embeddable_registry.ts

```diff
@@ -8,9 +8,54 @@
  */
 
 import { i18n } from '@kbn/i18n';
-import type { DefaultEmbeddableApi, EmbeddableFactory } from './types';
+import type {
+  ComposableFetchContextFactory,
+  DefaultEmbeddableApi,
+  EmbeddableFactory,
+} from './types';
 
 const registry: { [key: string]: () => Promise<EmbeddableFactory<any, any>> } = {};
+const composableFetchContextRegistry: {
+  [key: string]: () => Promise<ComposableFetchContextFactory<any>>;
+} = {};
+
+/**
+ * Registers a new composable fetch context factory. This should be called at plugin start time.
+ *
+ * @param type The key to register the factory under. This should match the key used to register the embeddable factory
+ * @param getFactory an async function that gets the factory definition for this key. This should always async import the
+ * actual factory definition file to avoid polluting page load.
+ */
+export const registerComposableFetchContextFactory = <SerializedState extends object = object>(
```

Comment:
@nreese I separated this into a completely different registry rather than adding it on to the existing embeddable factory registry.

This is because it felt wrong to me to need to pre-load the code for _every_ embeddable factory used on the dashboard just to check if it had a method on it for getting the composable fetch context or not.